core/shared/src/main/scala/org/http4s/multipart/PartReceiver.scala

```diff
@@ -0,0 +1,206 @@
+    part => Resource.eval(part.bodyText.compile.string).map(Right(_))
+
+  /** Creates a PartReceiver which writes the part body to a temporary file, then returns that file's `Path`. */
+  def toTempFile[F[_]](implicit F: Files[F], c: Compiler[F, F]): PartReceiver[F, Path] =
+    part =>
+      F.tempFile
+        .evalTap(path => part.body.through(F.writeAll(path)).compile.drain)
+        .map(Right(_))
+
+  /** Creates a PartReceiver that ignores the part body. */
+  def ignore[F[_]]: PartReceiver[F, Unit] =
+    _ => Resource.pure(Right(()))
+
+  /** Creates a PartReceiver that immediately returns the given `DecodeFailure` instead of consuming the part body.
+    *
+    * @param err The error returned by the created PartReceiver
+    */
+  def reject[F[_], A](err: DecodeFailure): PartReceiver[F, A] =
+    _ => Resource.pure(Left(err))
+
+  /** Creates a PartReceiver that delegates to an implicitly-resolved `EntityDecoder` to decode the part body,
+    * passing `strict = false` to the underlying `decode` method.
+    *
+    * This method passes `strict = false` to the `decoder.decode` method.
+    *
+    * @param decoder The delegate EntityDecoder
+    */
+  def decode[F[_], A](implicit decoder: EntityDecoder[F, A]): PartReceiver[F, A] =
+    part => Resource.eval(decoder.decode(part, strict = false).value)
+
+  /** Creates a PartReceiver that delegates to an implicitly-resolved `EntityDecoder` to decode the part body,
+    * passing `strict = true` to the underlying `decode` method.
+    *
+    * @param decoder The delegate EntityDecoder
+    */
+  def decodeStrict[F[_], A](implicit decoder: EntityDecoder[F, A]): PartReceiver[F, A] =
+    part => Resource.eval(decoder.decode(part, strict = true).value)
+
+  /** Creates a PartReceiver that loads the part's body into an in-memory buffer, up until
+    * a specified maximum size, at which point it will instead write the body to a temp file.
+    *
+    * The temp file created by this decoder (if any) will be released with the Resource returned
+    * by this receiver's `receive` method.
+    *
+    * @param maxSizeBeforeFile Maximum number of bytes to be buffered in memory for each part
+    *                          received by the returned receiver. When the in-memory buffer
+    *                          would exceed this limit, the buffer is dumped to a temp file
+    *                          and all subsequent data pulled from the part body will be
+    *                          appended to that file instead of accumulating the memory buffer.
+    * @param chunkSize The chunk size used when reading data back from a temporary file created by this receiver
+    * @return A PartReceiver which dynamically decides whether to buffer the part's body in memory or in a file
+    */
+  def toMixedBuffer[F[_]](
+      maxSizeBeforeFile: Int,
+      chunkSize: Int = 8192,
+  )(implicit
+      files: Files[F],
+      mt: MonadThrow[F],
+      c: Compiler[F, F],
```

Comment:
I've been warned by the fs2 authors to not take `Compiler` as a constraint.  Maybe it's just better to slap a `Concurrent` on `F`?